gui/storage/forms.py

```diff
@@ -130,6 +130,12 @@ def _clean_zfssize_fields(form, attrs, prefix):
     for attr in attrs:
         formfield = '%s%s' % (prefix, attr)
         match = r.match(cdata[formfield].replace(' ', ''))
+        try:
+            int(cdata[formfield])
+            continue
+        except ValueError:
+            pass
+
```

Comment:
That check looks wrong to me. cdata[formfield] may have letters, e.g. "10M" which will then be converted.